server/core/src/main/java/org/infinispan/server/core/CacheIgnoreManager.java

```diff
@@ -83,18 +88,38 @@ public void stop() {
    @Listener
    private final class CacheListener {
       @CacheEntryCreated
-      public void created(CacheEntryCreatedEvent<String, Set<String>> e) {
+      public void created(CacheEntryCreatedEvent<String, IgnoredCaches> e) {
          if (!e.isOriginLocal() && !e.isPre() && e.getKey().equals(IGNORED_CACHES_KEY)) {
             updateLocalCopy(e.getValue());
          }
       }
 
       @CacheEntryModified
-      public void modified(CacheEntryModifiedEvent<String, Set<String>> e) {
+      public void modified(CacheEntryModifiedEvent<String, IgnoredCaches> e) {
          if (!e.isOriginLocal() && !e.isPre() && e.getKey().equals(IGNORED_CACHES_KEY)) {
             updateLocalCopy(e.getValue());
          }
       }
    }
 
+   @ProtoTypeId(ProtoStreamTypeIds.IGNORED_CACHES)
+   static class IgnoredCaches {
+      final Set<String> caches;
+
+      IgnoredCaches() {
+         this(ConcurrentHashMap.newKeySet());
+      }
+
+      @ProtoFactory
+      IgnoredCaches(Set<String> caches) {
+         // ProtoStream cannot use KeySetView directly as it does not have a zero args constructor
+         this.caches = ConcurrentHashMap.newKeySet(caches.size());
+         this.caches.addAll(caches);
+      }
+
+      @ProtoField(number = 1, collectionImplementation = HashSet.class)
+      Set<String> getCaches() {
+         return caches;
+      }
+   }
```

Comment:
@gustavonalle @tristantarrant I had to introduce this wrapper class as ProtoStream cannot handle collections directly atm. The upside, is that we can use a TypeId so the payload is smaller and we can extend this entity in the future in a backwards compatible way if we ever need to store more information.